lib/stdlib/src/shell_docs_test.erl

```diff
@@ -0,0 +1,113 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 1996-2024. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%%
+-module(shell_docs_test).
+-moduledoc false.
+
+-include_lib("kernel/include/eep48.hrl").
+
+-export([module/2]).
+
+module(#docs_v1{ docs = Docs, module_doc = MD }, Bindings) ->
+    MDRes = parse_and_run(module_doc, MD, Bindings),
+    case lists:flatten(
+           [parse_and_run(KFA, EntryDocs, Bindings)
+            || {KFA, _Anno, _Sig, EntryDocs, _Meta} <- Docs, is_map(EntryDocs)] ++ MDRes) of
+        [] ->
+            ok;
+        Else ->
+            Else
+    end.
+
+parse_and_run(KFA, #{} = Ds, Bindings) ->
+    [parse_and_run(KFA, D, Bindings) || _ := D <- Ds];
```

Comment:
Not sure I understand this line. `Ds` was empty, what does `_ := D <- Ds` mean?
I am not even sure why `:=` is accepted or its meaning in a normal list.

Reply:
The match checks if Ds is a map, not if it is an empty map.